addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/launch-product/launch-product.js

```diff
@@ -356,6 +359,34 @@ class LaunchProduct extends StepBase {
     }
     return result.LaunchPathSummaries[0];
   }
+
+  /**
+   * Method to check if the resolved input parameter contained a static namespace param. If so, this method augments the
+   * namespace to begin with 'analysis-' for permissions purposes (if it does not already start with that) and to end with 
+   * a unique datetime string so Cloudformation doesn't make create duplicate stacks on seperate deployments of the same
+   * workspace configuration. 
+   * 
+   * @param resolvedInputParams 
+   * @param datetime 
+   * @returns {Promise<{Value: string, Key: string}[]>}
+   */
+  async checkNamespace(resolvedInputParams, datetime){
```

Comment:
This method has a side effect, in that it's changing `resolvedInputParams`. Let's change the method name to reflect that. `checkNamespace` => `updateNamespaceIfNecessary`.

Reply:
Yeah, I just learned the hard way and need to re-evaluate my logic in testing it because it does this. Is there a way to NOT make it do this? If it is easier to stop the changing of the parameter passed, I would prefer that so my tests would work *I think*. Thank you!

Reply:
We can always change `checkNamespace` method to `getNewNamespaceIfNecessary` to remove side effect. And just return the new namespace. In the start method we will have `resolvedInputParams[namespaceIndex].Value = getNewNamespaceIfNecessary(..`. For your unit tests you can just check the namespaces then and not need to look at resolvedInputParams.

Reply:
Re: only passing the namespace, I tried to keep most of the new logic in the new function so I only had to unit test the new function and not the start function. 

I think I will do Tim's method name change and just redo the logic in my unit tests as the renaming of the namespace works this way--the unit tests just don't because I didn't know about the side effect.